$NetBSD: patch-af,v 1.1 2007/03/18 21:24:00 rillig Exp $

Unfortunately, "case 1 ... 255" is not included in ISO C99, though it
would be very useful in some cases.

--- src/tui.c.orig	2001-02-11 20:28:53.000000000 +0100
+++ src/tui.c	2007-03-18 22:21:46.000000000 +0100
@@ -688,8 +688,8 @@ static void tui_player_control (struct T
     player_kill (p);
   
   if (!p->paused) {
-    switch (player_status (p)) {
-    case 0:
+    int status = player_status(p);
+    if (status == 0) {
       if (tui->queue->items && p->pid) {
 	queue_skip (tui->queue, tui->dinfo);
 	tui->disp[DM_QUEUE].items = tui->queue->items;
@@ -699,7 +699,7 @@ static void tui_player_control (struct T
 	if (!tui->queue->items)
 	  p->pid = 0;
       }
-    case 1 ... 255:
+    } else if (1 <= status && status <= 255) {
       if (tui->queue->items) {
 	item = queue_get_item (tui->queue, 0);
 	p->pid = player_play (tui->cfg, tui->dinfo, item);
